Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+JavaTranslation.swift

```diff
@@ -125,13 +164,100 @@ extension JNISwift2JavaGenerator {
     /// The name of the Java parent scope this function is declared in
     let parentName: String
 
-    /// Function signature
+    /// Function signature of the Java function the user will call
     let translatedFunctionSignature: TranslatedFunctionSignature
+
+    /// Function signature of the native function that will be implemented by Swift
+    let nativeFunctionSignature: NativeFunctionSignature
+  }
+
+  static func translate(knownType: SwiftKnownTypeDeclKind) -> JavaType? {
+    switch knownType {
+    case .bool: .boolean
+    case .int8: .byte
+    case .uint16: .char
+    case .int16: .short
+    case .int32: .int
+    case .int64: .long
+    case .float: .float
+    case .double: .double
+    case .void: .void
+    case .string: .javaLangString
+    case .int, .uint, .uint8, .uint32, .uint64,
+        .unsafeRawPointer, .unsafeMutableRawPointer,
+        .unsafePointer, .unsafeMutablePointer,
+        .unsafeRawBufferPointer, .unsafeMutableRawBufferPointer,
+        .unsafeBufferPointer, .unsafeMutableBufferPointer, .optional, .data, .dataProtocol:
+      nil
+    }
   }
 
   struct TranslatedFunctionSignature {
-    let parameters: [JavaParameter]
-    let resultType: JavaType
+    let selfParameter: TranslatedParameter?
+    let parameters: [TranslatedParameter]
+    let resultType: TranslatedResult
+
+    var requiresSwiftArena: Bool {
+      return self.resultType.conversion.requiresSwiftArena
+    }
+  }
+
+  /// Represent a Swift API parameter translated to Java.
+  struct TranslatedParameter {
+    let parameter: JavaParameter
+    let conversion: JavaNativeConversionStep
+  }
+
+  /// Represent a Swift API result translated to Java.
+  struct TranslatedResult {
+    let javaType: JavaType
+
+    /// Represents how to convert the Java native result into a user-facing result.
+    let conversion: JavaNativeConversionStep
+  }
```

Comment:
Hmmm, I wonder if we should strive to eventually share those types rather than dupe them 🤔 WDYT @rintaro

Reply:
Cool, let's keep this as is then :-) Yeah over-tying the two might cause more trouble than it's worth.